crates/torii/core/src/engine.rs

```diff
@@ -165,34 +156,28 @@ impl<'db, P: Provider + Sync + Send> Engine<'db, P> {
         Ok(())
     }
 
-    async fn process(&mut self, block: MaybePendingBlockWithTxs) -> Result<()> {
-        let block: BlockWithTxs = match block {
-            MaybePendingBlockWithTxs::Block(block) => block,
-            _ => return Ok(()),
-        };
-
-        Self::process_block(self, &block).await?;
+    async fn process(&mut self, event: EmittedEvent, current_block: &mut u64) -> Result<()> {
+        if event.block_number.unwrap() > *current_block {
+            *current_block = event.block_number.unwrap();
 
-        for (tx_idx, transaction) in block.clone().transactions.iter().enumerate() {
-            let transaction_hash = match transaction {
-                Transaction::Invoke(invoke_transaction) => {
-                    if let InvokeTransaction::V1(invoke_transaction) = invoke_transaction {
-                        invoke_transaction.transaction_hash
-                    } else {
-                        continue;
-                    }
-                }
-                Transaction::L1Handler(l1_handler_transaction) => {
-                    l1_handler_transaction.transaction_hash
-                }
-                _ => continue,
-            };
+            if let Some(ref block_tx) = self.block_tx {
+                block_tx.send(event.block_number.unwrap()).await?;
+            }
 
-            self.process_transaction_and_receipt(transaction_hash, transaction, &block, tx_idx)
+            Self::process_block(self, event.block_number.unwrap(), event.block_hash.unwrap())
                 .await?;
+            info!("processed block: {}", event.block_number.unwrap());
```

Comment:
```suggestion
            info!(target: "torii::engine", block = %event.block_number.unwrap(), "Processed block");
```

Let's update to structured logging. Something like this